content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1050 @@
+
+```rust
+fn system(mut players: Query<&mut Player>) {
+    for player in players.iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+Compare that to the complicated QuerySet that this would have needed in previous Bevy versions to avoid conflicting immutable and mutable Queries:
+
+```rust
+// Gross!
+fn system(mut players: QuerySet<(QueryState<&Player>, QueryState<&mut Player>)>) {
+    for player in players.q0().iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.q1().iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+### SystemState
+
+Have you ever wanted to use "system params" directly with a Bevy World? With {{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}}, now you can!
+
+```rust
+let mut system_state: SystemState<(Res<A>, Query<&B>)> = SystemState::new(&mut world);
+let (a, query) = system_state.get(&world);
+```
+
+{{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}} does all of the same caching that a normal Bevy system does, so reusing the same SystemState results in uber-fast World access.
+
+### Sub Apps
+
+<div class="release-feature-authors">authors: @cart, @zicklag, @bjorn3</div>
+
+The new Bevy renderer requires strict separation between the "main app" and the "render app". To enable this, we added the concept of "sub-apps":
+
+```rust
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, AppLabel)]
+pub struct RenderApp;
+
+let mut render_app = App::empty();
+app.add_sub_app(RenderApp, render_app, move |app_world, render_app| {
+    // execute app logic here
+});
+
+// later
+app.sub_app_mut(RenderApp)
+    .add_system(some_system);
+    .add_system(some_other_system);
+```
+
+We plan on exposing more control over scheduling, running, and working with sub-apps in the future.
+
+### Query::iter_combinations
+
+<div class="release-feature-authors">authors: @Frizi</div>
+
+You can now iterate all permutations of N entities for a given query:
+
+```rust
+fn system(query: Query<&Player>) {
+    // iterates each possible combination of two entities exactly once
+    for [p1, p2] in query.iter_combinations() {
+    }
+
+    // iterates each possible combination of three entities exactly once
+    for [p1, p2, p3] in query.iter_combinations() {
+    }
+}
+```
+
+This is especially useful for things like "checking for entities for collisions with all other entities". There is also an `iter_combinations_mut` variant. Just be careful ... the time complexity of this grows quickly (faster than exponentially!) as the number of entities in your combinations increases. With great power comes great responsibility!
+
+The new `iter_combinations.rs` example illustrates how to use this new API to calculate gravity between objects in a "solar system":
```

Comment:
there is https://github.com/bevyengine/bevy/pull/3075 that would make that screenshot less... gray